src/scheduler/work_bucket.rs

```diff
@@ -113,6 +113,9 @@ impl<VM: VMBinding> WorkBucket<VM> {
     pub fn add<W: GCWork<VM>>(&self, work: W) {
         self.add_with_priority(Self::DEFAULT_PRIORITY, Box::new(work));
     }
+    pub fn add_boxed(&self, work: Box<dyn GCWork<VM>>) {
```

Comment:
It is an easy refactoring. But it may not be worth it. If we only keep the 'boxed' version, we need to add `Box::new()` to ~27 callsites. In that case, you probably would like to extract the common pattern and have the unboxed version.